### DISCUSSION TITLE
Strawberry meeting #0

### DISCUSSION THREAD

**Discussion by jkimbo** [strawberry-graphql/strawberry#652] [Meeting Notes]
**Attendees:** @patrick91 @BryceBeagle @jkimbo 
**Date & time:** 2020-01-08 4pm (GMT)

## 🎯 Agenda

* Define a roadmap
* Decide on schedule for regular public meetings
* Newsletter
* Website (next+algolia)
 
### Roadmap
V1 roadmap outlined in: https://github.com/strawberry-graphql/strawberry/issues/268

Decided to de-prioritise permission support because there are lots of ways of implementing it and so it doesn't feel right that Strawberry should define only 1 way approach. Feel that it's better to have easy ways to implement permissions and strong examples on how to do it in different contexts. Implementing support for decorators (https://github.com/strawberry-graphql/strawberry/pull/473) might mean we don't have to have permissions as a first class concept in Strawberry.

Discussed the scope of the dev server: it's to provide an easy way to for people to get started. If you already have a web server then using the Strawberry integration usually makes more sense.

Decided that we're going to setup a new repo for more complicated examples than can fit in the docs. E.g. setting up subscriptions with Django ASGI.

Decided to de-prioritise pagination support for v1. Cursor pagination can probably come as part of Relay support and simple pagination (using page num and page size) could be part of the docs (example here: https://github.com/strawberry-graphql/strawberry/issues/175#issuecomment-632037277)

Discussed the status of the internal refactor that @BryceBeagle is leading: https://github.com/strawberry-graphql/strawberry/pull/489 
Update: still progressing but scope is quite large so will take some time. There are a couple of new features that are blocked by refactor (decorator support and implementing a `merge_types` function) but we're happy to wait until the refactor is done.

Discussion on how to present the roadmap and @patrick91 highlighted the Edge DB roadmap as a good example: https://www.edgedb.com/roadmap 
No decision made on actually implementing something similar but we all agreed with the intention.

### Regular public meetings
~Monthly. First one on a Saturday at the end of Jan. Time is TBD. Open to everyone.

### Newsletter
There are around 100 people subscribed currently.

@patrick91 to write the first post to cover what was achieved in 2020 and announcing @BryceBeagle and @jkimbo as core developers.

### Website
The website is being migrated to Next.js (from Gatsby). Currently the `main` branch has a mostly working version of it in Next.js. The reason for the migration was to move to server side rendering so that we can avoid a build step and fetch docs directly from Github. It will also make supporting versioned docs easier in the future.

Aim is to get the Next.js version of the site ready soon. We don't need to support versioning docs until after v1 but we should have search before v1.
﻿
## ﻿✅ Action items

* [x] @patrick91 to schedule the first public meeting.
* [x] @patrick91 to create a draft newsletter for Friday 15th.
* [ ] @patrick91 to send stickers next week.
* [x] @patrick91 to check the website this weekend.
* [x] @patrick91 to create an examples repo.
* [x] @BryceBeagle to finish schema conversion refactor by Jan ~17th: #662 
* [x] @jkimbo to write up these notes into Github discussions

**Comment by ar-to** [comment]
good work on these notes!  I look forward to helping out with what I can.

**Comment by ulgens** [comment]
There are a couple of notes what is being deprioritized, but I missed what is getting prioritized except decorators.

**Reply by patrick91** [reply]
Permissions -> these are already implemented, but personally I'm not sure they are good enough, I'll probably open a discussion around it in future
Pagination -> I would love to see built-in pagination, but I think we can do that for post v1, especially supporting relay :)

**Comment by patrick91** [comment]
Examples repo here: https://github.com/strawberry-graphql/examples
